Remove unused dependencies

aduth opened the pull request:
Related to ARC-127

## Proposed Changes

* Removes a number of presumably-unused dependencies

## Why are these changes being made?

* Improve install performance
* Reduce surface area for supply-chain attacks
* Reduce maintenance overhead

## Testing Instructions

* Verify that builds pass and application loads

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

aduth committed:
Remove unused dependencies

aduth commented on package.json:
```diff
@@ -102,7 +102,6 @@
 		"lint:config-defaults": "node bin/validate-config-keys.js",
 		"lint:css": "stylelint \"**/*.{css,scss}\"",
 		"lint:js": "eslint --ext .js,.jsx,.ts,.tsx,.mjs,.json --cache .",
-		"lint:mixedindent": "mixedindentlint --ignore-comments \"client/**/*.scss\" \"assets/**/*.scss\" \"**/*.js\" \"**/*.jsx\" \"**/*.tsx\" \"**/*.tsx\" \"!build/**\" \"!node_modules/**\" \"!public/**\" \"!client/config/index.js\"",
```
While this is still technically _usable_, I don't see it referenced in any CI tooling, so it may as well not exist. Additionally, I'd argue that we should be enforcing consistent indentation through tools like Prettier, ESLint, and Stylelint instead.

mirka replied:
⚠️ I think it's being run as part of the `"lint": "run-s -s 'lint:*'"` glob.

aduth commented on package.json:
```diff
@@ -246,7 +243,6 @@
 		"@babel/eslint-parser": "^7.26.10",
 		"@babel/register": "^7.25.9",
 		"@babel/runtime": "^7.26.10",
-		"@bartekbp/typescript-checkstyle": "^5.0.0",
```
Similarly, this and the `eslint-nibble` dependency below could be used in a developer's own terminal, but we don't have any documented workflows for it. I think these dependencies are the sort of thing where a developer could install them (optionally globally) in their own environment if they want to use them, rather than including them in the project by default.

tyxla replied:
Yep, I actually am still using these through `npx` all the time. There's very little value in keeping them in - maybe if we have some very special persisting config through a config file.

mirka commented on package.json:
```diff
@@ -230,7 +228,6 @@
 		"redux": "^5.0.1",
 		"request": "^2.88.2",
 		"swiper": "^10.1.0",
-		"workspace": "^0.0.1-preview.1",
```
Lol, seems like it was an unintended install (#83862), and a [very common](https://www.npmjs.com/package/workspace) one.

tyxla replied:
Good find!

mirka commented on package.json:
```diff
@@ -246,7 +243,6 @@
 		"@babel/eslint-parser": "^7.26.10",
 		"@babel/register": "^7.25.9",
 		"@babel/runtime": "^7.26.10",
-		"@bartekbp/typescript-checkstyle": "^5.0.0",
```
Made unnecessary in #60925.

mirka commented on package.json:
```diff
@@ -274,7 +270,6 @@
 		"dotenv-webpack": "^8.1.0",
 		"eslint": "^8.57.1",
 		"eslint-config-prettier": "^9.1.0",
-		"eslint-nibble": "^8.1.0",
```
Confirmed this is irrelevant for us right now (added in #44479).

mirka commented on package.json:
```diff
@@ -296,7 +291,6 @@
 		"jest-teamcity": "^1.12.0",
 		"loader-utils": "^1.4.2",
 		"lunr": "^2.3.9",
-		"mixedindentlint": "^1.2.0",
```
⚠️ Still used in `lint:*` glob.

aduth replied:
> ⚠️ Still used in `lint:*` glob.

Good catch! I still think my original point is valid, though as you commented maybe we can deal with that separately. I think if our current tooling already handles it, maybe it's safe to just include here? I'll double-check.

aduth replied:
I tested mixed indentation in a `.scss` file and it looks like our `yarn lint:css` doesn't currently catch it, so I'll restore this check (maybe limited to `.scss` files?). In a follow-up pull request, I'd suggest we incorporate Prettier into our Stylelint checks.

aduth replied:
Restored `mixedindentlint` in 78b425d0e5, limited to `.scss` files. I checked to confirm JavaScript files with mixed indentation are already flagged through ESLint (Prettier).

mirka commented on package.json:
```diff
@@ -210,7 +209,6 @@
 		"calypso": "workspace:^",
 		"calypso-codemods": "workspace:^",
 		"cmdk": "^0.2.1",
-		"config": "^3.3.12",
```
Confirmed this is no longer used. First added for Backstop tests in #52161.

mirka reviewed (approved):
Thanks for the cleanup! Confirmed they are all unused now except for `mixedindentlint`. Your point about preferring other ways to enforce indentation is valid though, but maybe that should be addressed in a separate PR.

aduth committed:
Restore mixedindentlint

Limited to scss files, where currently we don't have other mixed indentation checks. JavaScript files are checked through ESLint+Prettier

See: https://github.com/Automattic/wp-calypso/pull/102784#discussion_r2058918126

tyxla commented on package.json:
```diff
@@ -101,7 +101,7 @@
 		"lint:css": "stylelint \"**/*.{css,scss}\"",
 		"lint:js": "eslint --ext .js,.jsx,.ts,.tsx,.mjs,.json --cache .",
-		"lint:mixedindent": "mixedindentlint --ignore-comments \"client/**/*.scss\" \"assets/**/*.scss\" \"**/*.js\" \"**/*.jsx\" \"**/*.tsx\" \"**/*.tsx\" \"!build/**\" \"!node_modules/**\" \"!public/**\" \"!client/config/index.js\"",
+		"lint:mixedindent": "mixedindentlint --ignore-comments \"client/**/*.scss\" \"assets/**/*.scss\" \"!build/**\" \"!node_modules/**\" \"!public/**\"",
```
Good find with the `client/config/index.js` 👍
